src/main/java/com/direwolf20/buildinggadgets/common/network/split/PacketDecoder.java

```diff
@@ -0,0 +1,84 @@
+package com.direwolf20.buildinggadgets.common.network.split;
+
+import com.direwolf20.buildinggadgets.common.BuildingGadgets;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import io.netty.buffer.Unpooled;
+import net.minecraft.network.PacketBuffer;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+final class PacketDecoder<MSG> {
```

Comment:
Doesn't forge do a lot of this for you or is there just no support for packet splitting in forge?

Reply:
As far as I know not. Forge doesn't need that, so I don't know why they'd add it and I also didn't find it.